src/bootstrap/jobs/tracker_apis.rs

```diff
@@ -49,51 +51,56 @@ pub struct ApiServerJobStarted();
+    if config.enabled {
+        let bind_to = config
+            .bind_address
+            .parse::<std::net::SocketAddr>()
+            .expect("Tracker API bind_address invalid.");
 
-    // Run the API server
-    let join_handle = tokio::spawn(async move {
-        if !ssl_enabled {
-            info!("Starting Torrust APIs server on: http://{}", bind_addr);
+        let tls = make_rust_tls(config.ssl_enabled, &config.ssl_cert_path, &config.ssl_key_path)
+            .await
+            .map(|tls| tls.expect("tls config failed"));
 
-            let handle = server::start(bind_addr, tracker);
+        match version {
+            Version::V1 => Some(start_v1(bind_to, tls, tracker.clone()).await),
+        }
+    } else {
+        info!("Note: Not loading Http Tracker Service, Not Enabled in Configuration.");
+        None
+    }
+}
 
-            tx.send(ApiServerJobStarted()).expect("the API server should not be dropped");
+async fn start_v1(socket: SocketAddr, tls: Option<RustlsConfig>, tracker: Arc<core::Tracker>) -> JoinHandle<()> {
+    let server = ApiServer::new(Launcher::new(socket, tls))
+        .start(tracker)
+        .await
+        .expect("Failed to start Server");
 
-            if let Ok(()) = handle.await {
-                info!("Torrust APIs server on http://{} stopped", bind_addr);
-            }
-        } else if ssl_enabled && ssl_cert_path.is_some() && ssl_key_path.is_some() {
-            info!("Starting Torrust APIs server on: https://{}", bind_addr);
+    tokio::spawn(async move {
+        server.state.task.await.expect("failed to close service");
+    })
+}
 
-            let ssl_config = RustlsConfig::from_pem_file(ssl_cert_path.unwrap(), ssl_key_path.unwrap())
-                .await
-                .unwrap();
+#[cfg(test)]
+mod tests {
+    use std::sync::Arc;
 
-            let handle = server::start_tls(bind_addr, ssl_config, tracker);
+    use torrust_tracker_test_helpers::configuration::ephemeral_mode_public;
 
-            tx.send(ApiServerJobStarted()).expect("the API server should not be dropped");
+    use crate::bootstrap::app::initialize_with_configuration;
+    use crate::bootstrap::jobs::tracker_apis::start_job;
+    use crate::servers::apis::Version;
 
-            if let Ok(()) = handle.await {
-                info!("Torrust APIs server on https://{} stopped", bind_addr);
-            }
-        }
-    });
+    #[tokio::test]
+    async fn it_should_start_http_tracker() {
+        let cfg = Arc::new(ephemeral_mode_public());
+        let config = &cfg.http_api;
+        let tracker = initialize_with_configuration(&cfg);
+        let version = Version::V1;
 
-    // Wait until the APIs server job is running
-    match rx.await {
-        Ok(_msg) => info!("Torrust APIs server started"),
-        Err(e) => panic!("the API server was dropped: {e}"),
+        start_job(config, tracker, version).await.expect("some https server");
```

Comment:
Hi @da2ce7 there is a typo here:

```rust
    #[tokio::test]
    async fn it_should_start_the_tracker_api() {
        let cfg = Arc::new(ephemeral_mode_public());
        let config = &cfg.http_api;
        let tracker = initialize_with_configuration(&cfg);
        let version = Version::V1;

        start_job(config, tracker, version).await.expect("some tracker API server");
```

It should be the API, not the HTTP tracker.

Reply:
@josecelano thanks, and erros like it have also been fixed.